### PR TITLE
refactor: unify episode cards into shared EpisodeCard primitive

### LINKED ISSUE #344: refactor: unify episode cards into shared EpisodeCard primitive
## Context

Three near-duplicate episode card implementations have drifted visually:
- `src/components/notifications/notification-page-list.tsx` (inline `NotificationRow`) — no artwork, big score pill inline, minimal actions
- `src/components/podcasts/episode-card.tsx` — no artwork, tiny star+number score (custom `ScoreIndicator`), "Summarized/Failed" pill
- `src/components/library/saved-episode-card.tsx` — artwork tile, `WorthItBadge`, collapsible expansion

Design target: artwork tile (where appropriate) + podcast-title kicker + big title + description + topics, score pill top-right, divider, meta + actions row with primary `Listen` button.

## Scope

- Introduce `src/components/episodes/episode-card.tsx` primitive with props:
  `artwork?`, `podcastTitle`, `title`, `href`, `description?`, `topics?`, `score?`, `status?`, `meta`, `accent?: 'unread' | 'none'`, `primaryAction?`, `secondaryActions?`.
- Preserve the `isListened` prop + `ListenedButton` slot introduced by #335.
- Replace `Summarized/Failed` pill on podcast-page cards with the score pill (`WorthItBadge`); non-terminal/failure states become a small icon next to it. Score implies summarized.
- Delete `ScoreIndicator` helper in `podcasts/episode-card.tsx`.
- Migrate three call sites: `NotificationRow`, `podcasts/EpisodeCard`, `library/SavedEpisodeCard` (keeps its Collapsible wrapper).
- Notification card: artwork, primary `Listen` button, icon `AddToQueueButton`, icon `Dismiss`, `accent='unread'` when not read.
- Podcast-page card: no artwork, no primary button (whole card is a link), `ListenedButton` stays in secondary actions.
- Library saved card: artwork, no primary button, existing expand/remove/listened actions.
- Update VRT snapshots introduced by #335.

## Out of scope

- Using listen state as a visual accent (left bar) — tracked in follow-up issue, depends on this.
- Any data/schema changes.
- Audio-player wiring; primary `Listen` reuses existing play handling.

## Acceptance

- [ ] `<EpisodeCard>` primitive + Storybook stories covering prop combinations (artwork/no-artwork, unread/none, with/without primary action, listened/unlistened)
- [ ] Notification page visually matches the design target (screenshots in PR)
- [ ] Podcast detail page renders through the primitive; no `Summarized/Failed` pill; score pill present when scored; `ListenedButton` still works
- [ ] Library saved episodes render through the primitive
- [ ] VRT snapshots regenerated and reviewed
- [ ] `bun run lint && bun run test && bun run build` pass
- [ ] Browser-verified via `agent-browser` on notifications, a podcast page, and library

## Dependencies

Blocked on #335 (`feat: mark episode as already listened`) — this refactor touches the same files #335 modifies. Waiting for #335 to merge keeps the rebase cost on the structural PR, not on the feature PR.

### PR DESCRIPTION
**Pull request by funny-otter**
## Summary

- Introduce a single presentational primitive at `src/components/episodes/episode-card.tsx` with a slot-based API (`primaryAction`, `secondaryActions` as `ReactNode`). Props match the #344 spec: `artwork?`, `podcastTitle`, `title`, `href`, `description?`, `topics?`, `score?`, `status?`, `meta`, `accent?`, `primaryAction?`, `secondaryActions?`, `isListened?`.
- Migrate three drifted call sites through the primitive: `NotificationPageList` (artwork + Listen CTA + `AddToQueueButton` icon + dismiss icon + `accent='unread'` when not read), `podcasts/EpisodeCard` (thin adapter — no artwork, whole card link via outer `<Link>`, `ListenedButton` in secondary actions), `library/SavedEpisodeCard` (preserves outer `Collapsible` + `CollapsibleContent` expansion body).
- Replace the `Summarized`/`Failed` text pills on podcast cards with the `WorthItBadge` score pill plus a small inline `StatusIcon` (spinner for `queued`/`running`/`summarizing`, alert for `failed`, nothing for `completed`). Delete the bespoke `ScoreIndicator` helper.
- Preserve the `isListened` prop + `ListenedButton` slot from #335.
- Guard the notification primary `Listen` button so it only renders when `audioEpisode` is non-null; falls back to a `View episode` outline button otherwise.
- Storybook: add 14 `Episodes/EpisodeCard` stories (artwork × accent × primary-action × listened + processing/score/long-title/topics edge cases).
- Tests: 22 new unit tests for the primitive; existing call-site tests updated (notification list assertions now target the Listen button; podcast-card tests drop `Summarized`/`Failed` text assertions). Full suite: **2092/2092 passing**. `bun run lint` + `bun run build` clean.
- VRT: drop 17 stale baselines (`podcasts-episodecard--*`, `notifications-notificationpagelist--*`). Regeneration runs on CI dispatch on Linux/Chromium per ADR-024 — **do not regenerate locally on macOS**.

No new dependencies; no schema changes; no audio-player wiring changes.

## Test plan

- [x] `bun run lint` — 0 warnings
- [x] `bun run test` — 2092/2092 passing (unit + integration + component)
- [x] `bun run build` — Next.js production build succeeds
- [x] `bun run build-storybook` / `bun run storybook` — all 14 new `Episodes/EpisodeCard` stories render; existing stories still render
- [x] agent-browser — `/notifications` (unread + read tabs, dismiss flow, primary Listen button)
- [x] agent-browser — `/podcast/<id>` (score pill present, no `Summarized`/`Failed` text, `ListenedButton` works, whole-card link navigates to `/episode/<id>`)
- [x] agent-browser — `/library` (artwork, score pill, expand/collapse, remove, `MoveToCollection`, `ListenedButton`)
- [ ] CI VRT job regenerates baselines on Linux/Chromium dispatch and commits fresh PNGs
- [ ] Post-merge spot-check on preview deployment

## Plan + Review artifacts

Plan: `.dev/cdt/plans/plan-20260423-1853.md` — PM verdict APPROVED after one revision (T4 location field fix). Reviewer approved after one fix cycle (notification primary-action guarded by `audioEpisode`).

Closes #344

## Summary by CodeRabbit

* **New Features**
  * Consolidated episode card UI across notifications, podcasts, and library; notification rows show an in-place "Listen" action for direct playback (hidden when no episode audio/link)
  * Saved-library podcast titles now link correctly to podcast detail pages

* **UI Updates**
  * Summary/status pills replaced with compact icon-only indicators; score and topic displays unified; unread/listened accents preserved

* **Tests**
  * Added/updated tests and Storybook stories covering episode card variants

* **Documentation**
  * CHANGELOG updated documenting these UI/behavior changes

### CODE REVIEW

File: src/components/notifications/notification-page-list.tsx [hunk 1]
```diff
@@ -299,53 +298,62 @@ function NotificationRow({
...
+        podcastTitle={item.podcastTitle ?? ""}
+        title={item.episodeTitle ?? item.title}
+        href={
+          item.episodePodcastIndexId
+            ? ROUTES.episode(item.episodePodcastIndexId)
+            : ROUTES.DASHBOARD
+        }
+        topics={topics}
+        score={item.worthItScore}
+        accent={item.isRead ? "none" : "unread"}
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/notifications/notification-page-list.tsx:344]: Fixed: added `onTitleClick` prop to `EpisodeCard` primitive and pass `() => markReadOptimistic(item)` from `NotificationRow`. Title and artwork link clicks now flip local `isRead` via the same optimistic path the Listen/View buttons use. New regression test asserts clicking the title link calls `markNotificationRead`.

File: src/components/episodes/episode-card.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  /** Meta row cells rendered between description and action row. */
+  meta: ReactNode[];
+  /** Visual accent. `"unread"` applies the notification-unread bg tint. */
+  accent?: "unread" | "none";
+  /** Primary CTA shown in the action row as a full-text button (e.g. Listen). */
+  primaryAction?: ReactNode;
+  /** Icon-sized action slots (AddToQueueButton, ListenedButton, dismiss, remove, etc.). */
+  secondaryActions?: ReactNode;
+  /** When true, marks the card as listened — used for data attrs and future styling. */
+  isListened?: boolean;
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/episodes/episode-card.tsx:52]: Fixed: added `onTitleClick?: () => void` to `EpisodeCardProps` with a doc comment noting it runs alongside Link navigation.

File: src/components/episodes/episode-card.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  href,
+  description,
+  topics,
+  score,
+  status,
+  meta,
+  accent = "none",
+  primaryAction,
+  secondaryActions,
+  isListened = false,
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/episodes/episode-card.tsx:137]: Fixed: `onTitleClick` is now destructured in the function signature alongside the other props.

File: src/components/episodes/episode-card.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      data-listened={isListened}
+      className={cn(
+        "group transition-colors hover:bg-accent/50",
+        accent === "unread" && "bg-accent/10"
+      )}
+    >
+      <CardContent className="p-4">
+        <div className="flex gap-4">
+          {artwork !== undefined && (
+            <Link href={href} className="shrink-0">
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/episodes/episode-card.tsx]: Fixed: the artwork `<Link>` (inside `ArtworkTile`) now receives `onClick={onTitleClick}`. It is also marked `aria-hidden="true"` + `tabIndex={-1}` to avoid duplicate screen-reader announcements per CodeRabbit's a11y nitpick.

File: src/components/episodes/episode-card.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            </Link>
+          )}
+
+          <div className="min-w-0 flex-1">
+            <div className="flex items-start justify-between gap-2">
+              <div className="min-w-0 flex-1 space-y-0.5">
+                <p className="truncate text-xs text-muted-foreground">
+                  {podcastTitle}
+                </p>
+                <Link href={href} className="block">
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/episodes/episode-card.tsx]: Fixed: the title `<Link>` now receives `onClick={onTitleClick}` when an `href` is present.

File: src/components/podcasts/episode-card.tsx [hunk 6]
```diff
@@ -22,101 +19,89 @@ interface EpisodeCardProps {
...
+    <EpisodeCardPrimitive
+      podcastTitle={episode.feedTitle ?? "Podcast"}
+      title={episode.title}
+      href={`/episode/${episode.id}`}
+      description={
+        episode.description
+          ? stripHtml(episode.description)
+          : "No description available"
+      }
+      score={worthItScore ?? undefined}
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/podcasts/episode-card.tsx:100]: Answered: the split is intentional. Notifications and library are the new unified surfaces where 'Not rated' is valuable signal (user decides whether to click through based on score availability). The podcasts detail page renders a dense list where most episodes will be unsummarized by default, so showing 'Not rated' on every row would be noisy — hence the wrapper coerces `null` → `undefined` to hide the pill. The primitive's contract (`null` → 'Not rated', `undefined` → no badge) gives callers that choice; we exercise both sides deliberately.

**Comment by funny-otter** [comment]
> **Actionable comments posted: 1** — outside-diff mockPush assertions and podcastTitle fallback nit.

Fixed:
- `expect(mockPush).not.toHaveBeenCalled()` is now asserted on every Listen-click test path (unread, already-read, server-failure, and throw-path), plus the new title-click regression test covers the inverse (title click DOES mark as read, Listen click does NOT navigate).
- `podcastTitle: episode.feedTitle ?? "Podcast"` was already applied in the simplify pass (see commit 8b9b638) — it matches the `AddToQueueButton` fallback in the same wrapper.

**Comment by funny-otter** [comment]
> **Actionable comments posted: 1** — 3 nitpicks (library score handling, a11y sibling links, episode-number guard).

Fixed:
- **A11y (artwork sibling link):** `ArtworkTile` now renders its `<Link>` with `aria-hidden="true"` + `tabIndex={-1}`, so screen readers only encounter one primary 'Open episode' link per card (the title). Unit test `marks the artwork link as aria-hidden...` locks this in.
- **Episode-number guard:** `episode.episode !== null` → `typeof episode.episode === "number" && episode.episode > 0` to reject 0/undefined cleanly.

Answered (library score):
The library intentionally shows 'Not rated' for unrated episodes (unified with notifications per the primitive's new contract). The podcasts detail page is the outlier and preserves the legacy hide-when-null via `?? undefined` — see the sibling reply on the inline thread for the density rationale. The primitive's `null` → 'Not rated' / `undefined` → no-badge split is the lever that lets each caller pick its own behavior.

File: src/components/notifications/notification-page-list.tsx [hunk 7]
```diff
@@ -299,53 +306,63 @@ function NotificationRow({
...
+  return (
+    <article data-read={item.isRead}>
+      <EpisodeCard
+        artwork={item.artwork}
+        podcastTitle={item.podcastTitle ?? ""}
+        title={item.episodeTitle ?? item.title}
+        href={
+          item.episodePodcastIndexId
+            ? ROUTES.episode(item.episodePodcastIndexId)
+            : undefined
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/notifications/notification-page-list.tsx:339]: Fixed: `onTitleClick` was added to the `EpisodeCard` primitive in commit d52619d and is wired to both the title and artwork `<Link>` elements. `NotificationRow` passes `onTitleClick={() => onMarkRead(item)}`, so clicking the title or artwork now triggers the same optimistic `markReadOptimistic` path the View/Listen buttons use. A regression test `clicking the title link marks the notification as read` locks the behavior. This review was running against a stale SHA.

File: src/components/notifications/notification-page-list.tsx [hunk 8]
```diff
@@ -299,53 +306,63 @@ function NotificationRow({
...
-          onClick={() => onDismiss(item.id)}
-        >
-          <X className="h-4 w-4" />
-        </Button>
-      </div>
+  return (
+    <article data-read={item.isRead}>
+      <EpisodeCard
+        artwork={item.artwork}
+        podcastTitle={item.podcastTitle ?? ""}
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/notifications/notification-page-list.tsx]: Fixed: `item.podcastTitle ?? ""` → `item.podcastTitle ?? "Podcast"` in commit 4b694e5, so notifications with a null podcast-join now render a non-empty kicker. Matches the podcasts-detail wrapper and `AddToQueueButton` fallback.

File: src/components/library/saved-episode-card.tsx [hunk 9]
```diff
@@ -75,189 +76,155 @@ export function SavedEpisodeCard({ item, onRemoved, onCollectionChanged, isOffli
...
+      <EpisodeCard
+        artwork={podcast.imageUrl}
+        podcastTitle={podcast.title}
+        podcastHref={`/podcast/${podcast.podcastIndexId}?from=library`}
+        title={episode.title}
+        href={`/episode/${episode.podcastIndexId}`}
+        description={
+          episode.description ? stripHtml(episode.description) : undefined
+        }
+        score={episode.worthItScore}
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/library/saved-episode-card.tsx:180]: Answered: intentional — the library is one of two surfaces (notifications + library) where the unified primitive's `null` → 'Not rated' behavior is desired. Only the podcasts detail page preserves the legacy hide-when-null via `?? undefined` (dense episode list where most rows are unsummarized, so 'Not rated' on every row would be noise). The primitive's contract (`null` → 'Not rated', `undefined` → no badge) is the per-caller lever. See sibling replies on the CodeRabbit thread for the full rationale.

File: src/components/podcasts/episode-card.tsx [hunk 10]
```diff
@@ -22,101 +19,89 @@ interface EpisodeCardProps {
...
+    </>
   );
-}
 
+  return (
+    <EpisodeCardPrimitive
+      podcastTitle={episode.feedTitle ?? "Podcast"}
+      title={episode.title}
+      href={`/episode/${episode.id}`}
+      description={
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/podcasts/episode-card.tsx:95]: Answered: the PR description pre-dated the refactor. The final design intentionally links only the title + artwork inside the primitive to avoid nested-`<a>` invalid HTML that the legacy outer-`<Link>` pattern produced. Whole-card clickability is deferred per the PR handoff — we'll add it as an opt-in `fullCardHref` prop on the primitive if UX telemetry surfaces a need. The title-link regression is already covered by the new `clicking the title link marks the notification as read` test.

**Comment by funny-otter** [comment]
> **🧹 Nitpick comments (2)** — View-episode test mark-read assertion, and weak href=undefined assertion.

Fixed both in commit 4b694e5:
- The View-episode test now asserts `expect(mockMarkNotificationRead).toHaveBeenCalledWith(1)` alongside the existing `mockPush`/`mockPlayEpisode` expectations, locking the read-on-navigate side effect for the fallback path.
- The `renders title without a Link when href is omitted` test now asserts `container.querySelectorAll("a").toHaveLength(0)` and `titleNode.closest("a")` is `null` — the tautological `aria-label^="Open episode"` probe is gone. Nice catch.
